packages/orbit-components/src/Stack/helpers/getGap.js

```diff
@@ -5,18 +5,60 @@ import getSpacing from "./getSpacing";
 import getProperty from "./getProperty";
 import { QUERIES } from "../../utils/mediaQuery/consts";
 import type { GetGap } from "./getGap";
+import { rtlSpacing } from "../../utils/rtl";
+import { DIRECTIONS } from "../../utils/layout/consts";
+import type { Direction, Spacing } from "..";
+
+const getDirectionSpacingTemplate = (direction: Direction | Spacing): string => {
+  switch (direction) {
+    case DIRECTIONS.COLUMN:
+      return "0 0 __spacing__ 0";
+    case DIRECTIONS.ROW:
+      return "0 __spacing__ 0 0";
+    case DIRECTIONS.COLUMNREVERSE:
+      return "__spacing__ 0 0 0";
+    case DIRECTIONS.ROWREVERSE:
+      return "0 0 0 __spacing__";
+    default:
+      return "0 0 __spacing__ 0";
+  }
+};
+
+const isMobileViewport = viewport =>
+  viewport === "smallMobile" || viewport === "mediumMobile" || viewport === "largeMobile";
 
 const getGap: GetGap = ({ viewport, index, devices }) => props => {
   if (props[viewport] || viewport === QUERIES.DESKTOP) {
     const spacing = getProperty("spacing", { index, devices }, props);
     const direction = getProperty("direction", { index, devices }, props);
     const gap = spacing && direction && getSpacing(props)[spacing];
+    const isMobile = isMobileViewport(viewport);
+
+    const margin =
+      spacing &&
+      direction &&
+      String(getDirectionSpacingTemplate(direction)).replace(
+        "__spacing__",
+        getSpacing(props)[spacing],
+      );
 
     if (props.flex) {
       return css`
         gap: ${gap};
       `;
     }
+
+    return css`
+      & > * {
+        margin: ${margin && rtlSpacing(margin)}!important;
+        ${isMobile &&
+        css`
+          &:last-child {
+            margin: 0 !important;
```

Comment:
What is this for? Do we always want to add margin: 0 to the last child on mobile?

Reply:
yes, it's because last child does not need to have margin

Reply:
In general to the last child, but it was like that from the beginning, so not sure if we should change or keep that. We can change since the gap has the same behaviour. It's probably it was like that, for mobile, where usually is colomn direction and last element does not need to have a margin.